packages/gestalt/src/Tabs.js

```diff
@@ -73,45 +128,137 @@ function Tab({
+  index: number,
+  isActive: boolean,
+  onChange: OnChangeHandler,
+  size: 'md' | 'lg',
+|}) {
+  const [hovered, setHovered] = useState(false);
+  const [focused, setFocused] = useState(false);
+  const [pressed, setPressed] = useState(false);
+
+  let color = 'white';
+  if (pressed) {
+    color = 'lightWash';
+  } else if ((hovered || focused) && !isActive) {
+    color = 'lightGray';
+  }
+
+  return (
+    <Box id={id} paddingY={3}>
+      <TapArea
+        href={href}
+        onBlur={() => setFocused(false)}
+        onFocus={() => setFocused(true)}
+        onMouseDown={() => setPressed(true)}
+        onMouseUp={() => setPressed(false)}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+        onTap={({ event }) => onChange({ activeTabIndex: index, event })}
+        role="link"
+        rounding={TAB_ROUNDING}
+        tapStyle="compress"
+      >
+        <Flex alignItems="center" direction="column">
+          <Box
+            color={color}
+            padding={TAB_INNER_PADDING}
+            position="relative"
+            rounding={TAB_ROUNDING}
+            userSelect="none"
+          >
+            <Flex alignItems="center" gap={2} justifyContent="center">
+              <Text color="darkGray" overflow="noWrap" weight="bold">
+                {text}
+              </Text>
+
+              {indicator === 'dot' && <Dot />}
+              {/* Flow is dumb and doesn't realize Number.isFinite will return false for a string or undefined */}
+              {typeof indicator === 'number' && Number.isFinite(indicator) && (
+                <Count count={indicator} />
+              )}
+            </Flex>
+
+            {isActive && (
+              <Box
+                dangerouslySetInlineStyle={{ __style: { bottom: -UNDERLINE_HEIGHT } }}
+                position="absolute"
+                // 4px/boint, padding on left and right
+                width={`calc(100% - ${TAB_INNER_PADDING * 4 * 2}px)`}
+              >
+                <Underline />
+              </Box>
+            )}
+          </Box>
+        </Flex>
+      </TapArea>
+    </Box>
+  );
+}
+
 type Props = {|
   activeTabIndex: number,
   onChange: OnChangeHandler,
   size?: 'md' | 'lg',
-  tabs: $ReadOnlyArray<{|
-    href: string,
-    id?: string,
-    indicator?: 'dot',
-    text: Node,
-  |}>,
+  tabs: $ReadOnlyArray<TabType>,
   wrap?: boolean,
+  // Temporary prop for the Tabs Redesign project experiments
+  _dangerouslyUseV2?: boolean,
 |};
 
-export default function Tabs({ activeTabIndex, onChange, size = 'md', tabs, wrap }: Props): Node {
+export default function Tabs({
+  activeTabIndex,
+  onChange,
+  size = 'md',
+  tabs,
+  wrap,
+  _dangerouslyUseV2,
+}: Props): Node {
+  const TabComponent = _dangerouslyUseV2 ? TabV2 : Tab;
+
   return (
-    <Flex alignItems="center" justifyContent="start" wrap={wrap}>
+    <Flex alignItems="center" gap={_dangerouslyUseV2 ? 4 : 0} justifyContent="start" wrap={wrap}>
```

Comment:
handling all spacing between tabs here so we're not leaking that spacing to the Tabs component as a whole, allowing for tighter layouts on either side of the tabs

basically making it `TAB space TAB space TAB` instead of `space TAB space TAB space TAB space`